tools/codegen/diff_native_functions.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import os
+import requests  # TODO(jwtan): Figure out the conventional API to use within the PyTorch repository.
+import subprocess
+import sys
+import tempfile
+
+# TODO(jwtan): Figure out why we need this.
+root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, root)
+
+from tools.codegen.gen import parse_native_yaml
+
+# Returns the current_version and the base_version if specified by users.
+def get_versions() -> (str, str):
+    # TODO(jwtan): It returns a lot of information including release notes. Figure out a way to only query the tag names.
+    # TODO(jwtan): per_page=100 is not future proof.
+    response = requests.get("https://api.github.com/repos/pytorch/pytorch/releases?per_page=100")
```

Comment:
To me, having the script both ping GitHub network AND do diffing between native_functions.yaml feels like "doing too much". A more modular structure would be to scope this script to ONLY do diffing, and rely on some external mechanism for retrieving native_functions.yaml from older releases. This would, for example, make it easier to do testing, as you could just provide your own native_functions.yaml in those cases.